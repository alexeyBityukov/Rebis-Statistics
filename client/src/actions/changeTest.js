export const CHANGE_TEST = 'CHANGE_TEST';

export default () => ({
  type: CHANGE_TEST,
  reduxTest: 'flase',
});
