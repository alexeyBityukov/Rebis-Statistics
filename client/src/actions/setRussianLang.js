export const SET_RUSSIAN_LANG = 'SET_RUSSIAN_LANG';

export default () => ({
  type: SET_RUSSIAN_LANG,
});
