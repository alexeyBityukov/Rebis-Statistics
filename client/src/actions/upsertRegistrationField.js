export const UPSERT_REGISTRATION_FIELD = 'SET_RUSSIAN_LANG';

export default (fieldName, fieldValue) => ({
  type: UPSERT_REGISTRATION_FIELD,
  fieldName,
  fieldValue,
});
