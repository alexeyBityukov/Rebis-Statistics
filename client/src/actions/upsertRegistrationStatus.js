export const UPSERT_REGISTRATION_STATUS = 'UPSERT_REGISTRATION_STATUS';

export default status => ({
  type: UPSERT_REGISTRATION_STATUS,
  status,
});
