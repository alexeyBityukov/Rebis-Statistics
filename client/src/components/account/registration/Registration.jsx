import React from 'react';
import { withStyles, createStyles } from '@material-ui/core';
import Paper from '@material-ui/core/Paper';
import Button from '@material-ui/core/Button';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import ErrorMessage from '../../messages/ErrorMessage';
import TextField from '../../textFields';


const listFields = {
  email: 'email',
  password: 'password',
  repeatPassword: 'repeatPassword',
};

const styles = ({ spacing }) => createStyles({
  paper: {
    margin: spacing.unit,
    padding: spacing.unit,
  },
});

const validate = fields => (values) => {
  const errors = {};
  Object.values(fields).forEach((key) => {
    if (!values[key]) errors[key] = 'requiredField';
  });
  if (values.email && !/^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(values.email)) errors.email = 'invalidEmail';

  if (values.password !== values.repeatPassword) {
    errors.password = 'differentPasswords';
    errors.repeatPassword = 'differentPasswords';
  }

  return errors;
};

export default connect(
  state => ({ lang: state.main.lang }),
)(withStyles(styles)(reduxForm(
  {
    form: 'registration',
    validate: validate(listFields),
  },
)((props) => {
  const {
    classes,
    lang: { registration },
    handleSubmit,
    errorMessage,
  } = props;
  return (
    <Paper className={classes.paper}>
      <form autoComplete="off" onSubmit={handleSubmit}>
        <Field
          name={listFields.email}
          required
          label={registration.emailFieldLabel}
          component={TextField}
        />
        <Field
          name={listFields.password}
          required
          label={registration.passwordFieldLabel}
          component={TextField}
        />
        <Field
          name={listFields.repeatPassword}
          required
          label={registration.repeatPasswordFieldLabel}
          component={TextField}
        />
        <ErrorMessage label={errorMessage} />
        <Button
          color="primary"
          type="submit"
        >
          {registration.registerButtonLabel}
        </Button>
      </form>
    </Paper>
  );
})));
