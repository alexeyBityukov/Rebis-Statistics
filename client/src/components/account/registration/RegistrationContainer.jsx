import React, { Component } from 'react';
import Registration from './Registration';

export default class extends Component {
  state = {
    fields: {
    }
  };

  handleOnChange = (fieldName) => {
    return (event) => {
      event.preventDefault();
      this.setState({
        fields: {
          ...this.state.fields,
          [fieldName]: event.target.value,
        }
      });
    }
  };

  handleOnClick = (event) => {
    event.preventDefault();
    console.log('send req on save');
    console.log(this.state);
  };

  render() {
    return <Registration
      handleOnChange={this.handleOnChange}
      handleOnClick={this.handleOnClick}
      fields={this.state.fields}
    />;
  }
};
