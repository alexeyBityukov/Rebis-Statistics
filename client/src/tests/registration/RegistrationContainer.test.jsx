import React from 'react';
import { configure, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import { RegistrationContainer } from '../../components/account/registration/RegistrationContainer';
import { UPSERT_REGISTRATION_STATUS } from '../../actions/upsertRegistrationStatus';


configure({ adapter: new Adapter() });

const mockStore = configureMockStore();

let component;
let store;
const apiCreateUserErrorMessage = 'creatingUserError';

beforeEach(() => {
  store = mockStore({});
  component = shallow((<RegistrationContainer
    store={store}
    apiCreateUser={apiCreateUserErrorMessage}
  />));
});


const mockFetch = (status, statusText, response) => {
  window.fetch = jest.fn().mockImplementation(
    () => Promise.resolve(
      new window.Response(
        response,
        {
          status,
          statusText,
          headers: {
            'Content-type': 'application/json',
          },
        },
      ),
    ),
  );
};

describe('<RegistrationContainer>', () => {
  it('should shallow render correctly', () => {
    expect(shallowToJson(component)).toMatchSnapshot();
  });

  const submitValues = {
    username: 'username',
    password: 'password',
    repeatPassword: 'password',
  };

  it('call fetch to registration new user', async () => {
    mockFetch(200);
    await await component.instance().handleSubmit(submitValues);
    expect(store.getActions()).toEqual([{ type: UPSERT_REGISTRATION_STATUS, status: true }]);
  });

  it('should error, on call fetch to registration new user', async () => {
    mockFetch(400);
    await await component.instance().handleSubmit(submitValues);
    expect(store.getActions()).toEqual([{ type: UPSERT_REGISTRATION_STATUS, status: false }]);
    expect(component.state().errorMessage).toEqual(apiCreateUserErrorMessage);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
