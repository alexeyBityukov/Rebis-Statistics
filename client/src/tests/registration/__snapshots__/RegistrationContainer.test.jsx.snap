// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegistrationContainer> should error, on call fetch to registration new user 1`] = `
<Connect(WithStyles(ReduxForm))
  errorMessage="creatingUserError"
  onSubmit={[Function]}
/>
`;

exports[`<RegistrationContainer> should shallow render correctly 1`] = `
<Connect(WithStyles(ReduxForm))
  errorMessage=""
  onSubmit={[Function]}
/>
`;
